pypdf/_xobj_image_helpers.py

```diff
@@ -292,10 +290,17 @@ def _handle_jpx(
         mode = "RGBA"
     # we need to convert to the good mode
     try:
-        if img1.mode != mode:
+        if (img1.mode == mode) or (img1.mode in ("L", "P") and mode in ("L", "P")):
+            img = img1
+        elif (
+            img1.mode == "RGBA"
+            and mode == "CMYK"
+            or img1.mode == "CMYK"
+            and mode == "RGBA"
+        ):
```

Comment:
In theory, this whole condition is equivalent to `{img1.mode, mode} == {"CMYK", "RGBA"}`, but I do not know if this is understandable enough.

Reply:
I like it. Will add a comment to make it clearer